### PR TITLE
feat(argo-cd): Support for additional labels

### PR DESCRIPTION
**Pull request by nlamirault**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/redis/deployment.yaml [hunk 1]
```diff
@@ -6,7 +6,6 @@ metadata:
   name: {{ template "argo-cd.redis.fullname" . }}
   labels:
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.redis.name "name" .Values.redis.name) | nindent 4 }}
-    app.kubernetes.io/version: {{ .Values.redis.image.tag | quote }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/redis/deployment.yaml:9]: Please restore the version label change. Redis uses its own version which does not belong to the chart version

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argo-cd/templates/redis/deployment.yaml (hunk 1 above)
Comment (nlamirault) [reply charts/argo-cd/templates/redis/deployment.yaml]: done.

File: charts/argo-cd/templates/_helpers.tpl [hunk 2]
```diff
@@ -137,6 +137,10 @@ helm.sh/chart: {{ include "argo-cd.chart" .context }}
 {{ include "argo-cd.selectorLabels" (dict "context" .context "component" .component "name" .name) }}
 app.kubernetes.io/managed-by: {{ .context.Release.Service }}
 app.kubernetes.io/part-of: argocd
+app.kubernetes.io/version: {{ .context.Chart.AppVersion | quote }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/_helpers.tpl:140]: We cannot use the version in the template function. The version label is handled separately in every deployment.yaml

File: charts/argo-cd/templates/_helpers.tpl [hunk 3]
```diff
@@ -137,6 +137,10 @@ helm.sh/chart: {{ include "argo-cd.chart" .context }}
...
+{{- if .context.Values.global.additionalLabels }}
+{{ toYaml .context.Values.global.additionalLabels }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/_helpers.tpl:142]:
```suggestion
{{- with .context.Values.global.additionalLabels }}
{{ toYaml . }}
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Additional to the comments below, I'd like to ask if `global.podLabels` would solve your need. This is already present in code.

**Review by mbevc1 — APPROVED** [review]

**Review by mbevc1 — CHANGES_REQUESTED** [review]
As posted before we already have parameter that could be used.

**Comment by mkilchhofer** [comment]
> As posted before we already have parameter that could be used.

There might be a use case for this but idk😅.
The parameter we already have is only on the PodSpec. This feature here would be generic for all object metadata.

**Comment by mbevc1** [comment]
Ah, goatcha :)

**Review by mbevc1 — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**